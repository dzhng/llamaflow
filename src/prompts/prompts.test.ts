import { z } from 'zod';

import buildBooleanPrompt from './boolean';
import buildBulletPointsPrompt from './bulletPoints';
import buildJSONPrompt from './json';

describe('Prompt types', () => {
  it('Should build bullet prompt correctly', () => {
    expect(
      buildBulletPointsPrompt({
        message: 'What are the meanings of life?',
      }),
    ).toMatchSnapshot();

    expect(
      buildBulletPointsPrompt({
        message: 'What are the meanings of life?',
        amount: 10,
      }),
    ).toMatchSnapshot();

    expect(
      buildBulletPointsPrompt({
        message: 'What are the meanings of life?',
        amount: 3,
        length: 140,
      }),
    ).toMatchSnapshot();
  });

  it('Should build boolean prompt correctly', () => {
    expect(
      buildBooleanPrompt({
        message: 'The iPhone 14 was released in 2012',
        promptRetries: 2,
      }),
    ).toMatchSnapshot();
  });

  it('Should build JSON prompt correctly', () => {
    expect(
      buildJSONPrompt({
        initialMessage: 'What are some good baby names?',
        formatMessage: 'Respond as a JSON array, where each element in the array is one name.',
        schema: z.array(z.string()),
      }),
    ).toMatchSnapshot();
  });
});
