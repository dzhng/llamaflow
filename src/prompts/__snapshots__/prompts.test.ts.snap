// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt types Should build JSON prompt correctly 1`] = `
{
  "message": "What are some good baby names?
Respond as a JSON array, where each element in the array is one name.",
  "parse": [Function],
}
`;

exports[`Prompt types Should build boolean prompt correctly 1`] = `
{
  "message": "The iPhone 14 was released in 2012
Respond to the above statement only with the word "true" or "false", nothing else.",
  "parse": [Function],
}
`;

exports[`Prompt types Should build bullet prompt correctly 1`] = `
{
  "message": "What are the meanings of life?
Respond in bullet points, where each bullet point starts with the - character. Don't include any other text other than the bullet points.",
  "parse": [Function],
}
`;

exports[`Prompt types Should build bullet prompt correctly 2`] = `
{
  "message": "What are the meanings of life?
Respond in bullet points, where each bullet point starts with the - character. Don't include any other text other than the bullet points. There should be exactly 10 bullet points, no more or less.",
  "parse": [Function],
}
`;

exports[`Prompt types Should build bullet prompt correctly 3`] = `
{
  "message": "What are the meanings of life?
Respond in bullet points, where each bullet point starts with the - character. Don't include any other text other than the bullet points. Each bullet point should be less than 140 characters long, including white spaces. There should be exactly 3 bullet points, no more or less.",
  "parse": [Function],
}
`;
